### PR TITLE
Add new hats and preprocessing script

### LINKED ISSUE #269: Do one more pass at shapes and colors
- [ ] `eye`: slightly wider pupil?
- [ ] `frame`: with a bit more paint
- [ ] `curve`: slightly thicker
- [ ] Tweak light colors
- [x] Maybe make yellow darker
- [x] New hole
- [x] Maybe make pink brighter
- [x] Try not to clip descenders from line above
- [x] Tweak alignment

### PR DESCRIPTION
**Pull request by pokey**
image

image

The above screenshots are with SF Mono and the following configuration:

```json
{
     "cursorless.hatSizeAdjustment": 20,
     "cursorless.hatVerticalOffset": 5
}
```

Closes #269

- [x] Rename `ex` => `x`?
- [x] Rename `star` => `cross`
- [ ] Make `bolt` a little prettier
- [ ] Change aspect ratios of `frame` and `hole` to make them easier to tell apart
- [ ] Bend green and blue slightly further away from each other
- [x] Fatten `wing`
- [x] Tweak sizes to avoid clipping
- [x] Tweak alignment
- [x] Extra credit: hole and star tough to distinguish
- [x] `ex`: reduce width
- [x] `curve`: use flipped fox without point
- [x] `yellow`: darken on light mode
- [x] `gray`: tweak on light mode
- [x] `eye`: slightly wider pupil
- [x] `frame`: with a bit more paint
- [x] `wing`: potentially raise center without changing location of points
- [x] Remove useless aspect ratio stuff
- [x] Normalize icons by height instead of width?
- [x] Align hats
- [x] Maybe make pink brighter
- [x] Try not to clip descenders from line above

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -236,16 +236,20 @@
...
+						},
+						"play": {
+							"type": "boolean"
 						}
 					},
 					"default": {
-						"fourPointStar": true,
-						"chevron": true,
+						"fourPointStar": false,
+						"chevron": false,
```
Comment (pokey) [inline package.json:246]: Turning these off by default

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Contributing
```
Comment (pokey) [inline CONTRIBUTING.md:1]: Moving contributing info to its own file

File: src/scripts/preprocessSvgHats.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import * as parser from "fast-xml-parser";
```
Comment (pokey) [inline src/scripts/preprocessSvgHats.ts:1]: Got tired of doing this stuff manually

**Comment by AndreasArvidsson** [comment]
Now that our SVGs actually have padding on top and bottom we should definitely normalize on height

File: package.json [hunk 4]
```diff
@@ -228,13 +228,13 @@
 					"description": "Which shapes to enable",
 					"type": "object",
 					"properties": {
-						"fourPointStar": {
+						"ex": {
```
Comment (pokey) [inline package.json:231]: Changing these to make them easier to remember

File: package.json [hunk 5]
```diff
@@ -314,23 +322,72 @@
...
+						"play": 1,
+						"star": 1
 					},
 					"additionalProperties": false
 				},
 				"cursorless.individualHatAdjustments": {
 					"description": "Separate adjustments for each hat shape",
 					"type": "object",
-					"default": {}
+					"default": {
```
Comment (pokey) [inline package.json:349]: Put a default value in to make this one easier to use when tweaking.  Tempted to remove this stuff but it is super useful

File: src/core/Decorations.ts [hunk 6]
```diff
@@ -238,34 +198,40 @@ export default class Decorations {
...
 
     const svgWidthPx = Math.ceil(characterWidth);
-    const svgHeightPx = characterHeight + hatHeightPx + hatVerticalOffset;
+    const svgHeightPx = characterHeight + hatHeightPx + hatVerticalOffsetPx;
 
-    const newViewBoxWidth =
-      ((originalViewBoxWidth / hatWidthToCharacterWidthRatio) * svgWidthPx) /
-      characterWidth;
-    const newViewBoxHeight = (newViewBoxWidth * svgHeightPx) / svgWidthPx;
+    const newViewBoxWidth = originalViewBoxWidth * (svgWidthPx / hatWidthPx);
```
Comment (pokey) [inline src/core/Decorations.ts:233]: This one simplified nicely

File: src/scripts/addHatAdjustments.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (pokey) [inline src/scripts/addHatAdjustments.ts:1]: Got tired of doing this one manually as well

File: images/hats/bolt.svg [hunk 8]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 377.73 272.77" width="0.7em" height="0.7em"><defs><style>.cls-1{fill:black;}</style></defs><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><path class="cls-1" d="M377.73,48V0s-84.23,168.46-125,162.43c-31.64-4.68-113.06-108.7-113.06-108.7L6,214.53,0,272.77S99.58,118.5,141,126.22C171.82,132,262,242.79,262,242.79Z"/></g></g></svg>
```
Comment (pokey) [inline images/hats/bolt.svg:1]: This one could use a little tweaking, but I think it's good enough to ship for now

File: src/core/exposeEnablements.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import { join } from "path";
```
Comment (pokey) [inline src/core/exposeEnablements.ts:1]: @AndreasArvidsson Ok I switched to exposing state via file.  Just copied some basic stuff from command server; should be fairly robust

**Review by AndreasArvidsson — APPROVED** [review]

**Pull request merged**